config/puma.rb

```diff
@@ -42,3 +42,12 @@
 
 # Allow puma to be restarted by `rails restart` command.
 plugin :tmp_restart
+
+pidfile "/tmp/puma.pid"
+state_path "/tmp/puma.state"
+
+if ENV["VELUM_PORT"].to_i == 443
+  ssl_bind "0.0.0.0", ENV["VELUM_PORT"], key: "/etc/pki/velum.key", cert: "/etc/pki/velum.crt"
```

Comment:
Re 0.0.0.0 - I think, in this particular instance, we should be OK with this (caveat below..). This specific bind is intended to be world accessible, so 0.0.0.0 doesn't bind us to any networks we consider less secure than we intended to bind to.

Caveat: We should be OK with this, assuming it's nontrivial to reliably find the correct bind address here, as we gain TLS.

In our case, this is non-trivial to find. Given a machine with 2 NICs, we don't know which one to expose the public interfaces on. We could assume the first NIC, but this will undoubtedly be wrong in many scenarios. What we would need to do, is ask at install time what NICs should be used for which purposes, but that itself relies on a better network architecture.